src/value/borrowed.rs

```diff
@@ -91,221 +80,251 @@ impl<'v> Value<'v> {
 impl<'v> ValueTrait for Value<'v> {
     type Key = Cow<'v, str>;
 
+    #[inline]
+    fn array() -> Self {
+        Self::Array(Vec::new())
+    }
+    #[inline]
+    fn object() -> Self {
+        Self::Object(Box::new(Object::new()))
+    }
+    #[inline]
+    fn null() -> Self {
+        Self::Static(StaticNode::Null)
+    }
+
+    #[inline]
     fn value_type(&self) -> ValueType {
         match self {
-            Value::Null => ValueType::Null,
-            Value::Bool(_) => ValueType::Bool,
-            Value::F64(_) => ValueType::F64,
-            Value::I64(_) => ValueType::I64,
-            Value::String(_) => ValueType::String,
-            Value::Array(_) => ValueType::Array,
-            Value::Object(_) => ValueType::Object,
+            Self::Static(StaticNode::Null) => ValueType::Null,
```

Comment:
🎉

Reply:
that actually could be simplified away :) since value_type is implemented for StaticNode. Good catch!